### DISCUSSION TITLE
[Error: [appCheck/token-error] com.google.android.play.core.integrity.IntegrityServiceException: -8: Integrity API error (-8): The calling app is making too many requests to the API and hence is throttled

### DISCUSSION THREAD

**Discussion by fidodido75** [invertase/react-native-firebase#7040] [Q&A] [answered]
Migrated to Play Integrity from SafetyNet, but we are experiencing occasional 'Too Many Attempts' error from the Play Integrity mainly whilst trying to registering/updating a device. Our `getDevice` method is as below, we get the `fcmToken` first, then the `appCheckToken `and the `deviceId` and use these to get the device Response:

`export async function getDevice(): Promise<AxiosResponse<DeviceResponse>> {
  const fcmToken = await messaging().getToken();
  const appCheckToken = await appCheck().getToken(true);
  const deviceId = getUniqueId();
  const signature = await CryptoJS.SHA256(`${fcmToken}${deviceId}`).toString();
  return client.get<DeviceResponse>(
    `${BASE_URL}/0/devices/${deviceId}`,
    {
      headers: {
        'x-app': signature,
        'X-Firebase-AppCheck': appCheckToken.token,
      },
    },
  );
}`
The error does not crash the app and when you retry it works as expected when you go back and try again. But it's annoying to have a product working intermittently. There is a warning for not obtaining the integrity token too frequently as with the logic of AppCheck and we have not changed how it worked. 
Any ideas what could be triggering that `too many attempts` error within the Play Integrity API?
(using @react-native-firebase/app-check 17.4.1, react-native": "0.67.5")

**Comment by fidodido75** [comment, marked as answer]
I think this is due to setting the forceRefresh parameter to true. Still testing but seems to have cleared the error for now.

**Comment by deekshi0123-pixel** [comment]
Error getting token from Google
